@yongtang  Here are the internal errors, can you please verify ?

Traceback (most recent call last):
  File "/third_party/tensorflow/python/framework/ops.py", line 1840, in _create_c_op
    c_op = pywrap_tf_session.TF_FinishOperation(op_desc)
third_party.tensorflow.python.framework.errors_impl.InvalidArgumentError: Shape must be at least rank 1 but is rank 0 for '{{node SegmentMean}} = SegmentMean[T=DT_UINT16, Tindices=DT_INT64](SegmentMean/data, SegmentMean/segment_ids)' with input shapes: [], [0].

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/third_party/tensorflow/python/framework/test_util.py", line 2963, in assertRaisesWithPredicateMatch
    yield
  File "/third_party/tensorflow/python/kernel_tests/segment_reduction_ops_test.py", line 264, in testDataInvalid
    data=np.uint16(10), segment_ids=np.array([]).astype("int64"))
  File "/third_party/tensorflow/python/ops/gen_math_ops.py", line 8403, in segment_mean
    "SegmentMean", data=data, segment_ids=segment_ids, name=name)
  File "/third_party/tensorflow/python/framework/op_def_library.py", line 776, in _apply_op_helper
    attrs=attr_protos, op_def=op_def)
  File "/third_party/tensorflow/python/framework/ops.py", line 3519, in _create_op_internal
    op_def=op_def)
  File "/third_party/tensorflow/python/framework/ops.py", line 2003, in __init__
    control_input_ops, op_def)
  File "/third_party/tensorflow/python/framework/ops.py", line 1843, in _create_c_op
    raise ValueError(str(e))
ValueError: Shape must be at least rank 1 but is rank 0 for '{{node SegmentMean}} = SegmentMean[T=DT_UINT16, Tindices=DT_INT64](SegmentMean/data, SegmentMean/segment_ids)' with input shapes: [], [0].

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<embedded stdlib>/unittest/case.py", line 59, in testPartExecutor
    yield
  File "<embedded stdlib>/unittest/case.py", line 605, in run
    testMethod()
  File "/third_party/tensorflow/python/kernel_tests/segment_reduction_ops_test.py", line 265, in testDataInvalid
    self.evaluate(s)
  File "<embedded stdlib>/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/third_party/tensorflow/python/framework/test_util.py", line 2968, in assertRaisesWithPredicateMatch
    (str(type(e)), str(e)))
AssertionError: Exception of type <class 'ValueError'>: Shape must be at least rank 1 but is rank 0 for '{{node SegmentMean}} = SegmentMean[T=DT_UINT16, Tindices=DT_INT64](SegmentMean/data, SegmentMean/segment_ids)' with input shapes: [], [0].

--- a/tensorflow/core/kernels/segment_reduction_ops_impl_1.cc
+++ b/tensorflow/core/kernels/segment_reduction_ops_impl_1.cc
@@ -22,6 +22,8 @@ namespace internal {
 void SegmentReductionValidationHelper(OpKernelContext* context,
                                       const Tensor& input,
                                       const Tensor& segment_ids) {
+  OP_REQUIRES(context, TensorShapeUtils::IsVectorOrHigher(input.shape()),
+              errors::InvalidArgument("input must be at least rank 1"));
   OP_REQUIRES(context, TensorShapeUtils::IsVector(segment_ids.shape()),
               errors::InvalidArgument("segment_ids should be a vector."));
   const int64 num_indices = segment_ids.NumElements();

--- a/tensorflow/python/kernel_tests/segment_reduction_ops_test.py
+++ b/tensorflow/python/kernel_tests/segment_reduction_ops_test.py
@@ -25,6 +25,7 @@ import numpy as np
 from tensorflow.python.client import session
 from tensorflow.python.framework import constant_op
 from tensorflow.python.framework import dtypes as dtypes_lib
+from tensorflow.python.framework import errors_impl
 from tensorflow.python.framework import ops
 from tensorflow.python.framework import test_util
 from tensorflow.python.ops import gradient_checker
@@ -254,6 +255,17 @@ class SegmentReductionOpTest(SegmentReductionHelper):
             x_init_value=np_x.astype(np.double),
             delta=1)
       self.assertAllClose(jacob_t, jacob_n)
+
+  def testDataInvalid(self):
+    # Test case for GitHub issue 40653.
+    for use_gpu in [True, False]:
+      with self.cached_session(use_gpu=use_gpu):
+        with self.assertRaisesRegex(
+            (ValueError, errors_impl.InvalidArgumentError),
+            "must be at least rank 1"):
+          s = math_ops.segment_mean(
+              data=np.uint16(10), segment_ids=np.array([]).astype('int64'))
+          self.evaluate(s)
 
 
 class UnsortedSegmentTest(SegmentReductionHelper):

